.github/workflows/flake-maintenance.yml

```diff
@@ -0,0 +1,33 @@
+name: Flake maintenance
+
+on:
+  schedule:
+    - cron: "0 14 * * 1" # weekly at 14:00 UTC
+  workflow_dispatch:
+
+jobs:
+  update-flake-lock:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@main
+
+      - name: Enable magic Nix cache
+        uses: DeterminateSystems/magic-nix-cache-action@main
+
+      - name: Update flake.lock
+        uses: DeterminateSystems/update-flake-lock@main
+        with:
+          pr-title: "chore: update flake.lock"
+          git-commit-message: "chore: update flake.lock"
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run full flake check (updated lock)
+        run: nix flake check -L
```

Comment:
is this because the flake check on the PR it opens is planned to be partial?

if so maybe it's better to do it in the reverse order:

First:

```sh
nix flake check -L --recreate-lock-file --no-write-lock-file
```

to make a new lockfile

then add `--offline` to the update-flake-lock action's nix arguments, if it passes this to `nix flake update --commit-lockfile` then i think it will commit without re-fetching the upstream flake inputs so the same flake inputs as were used when running the checks with the recreated lock file should be used... a cleaner approach would be to PR to the update-flake-lock action to allow it to run the check after the `nix flake update` command but before opening the PR